File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn get-freqs
  "Count letters in word"
  [word]
  (->
    (str/lower-case word)
    (frequencies)))

(defn same?
  [a b]
  (=
    (str/lower-case a)
    (str/lower-case b)))

(defn anagrams-for
  "Find anagrams of a word in list"
  [word anagrams]
  (let [freqs (get-freqs word)]
    (filter (fn [w]
      (and
        (= (get-freqs w) freqs)
        (not (same? w word))))
      anagrams)))

File: haskell/anagram/Anagram.hs
module Anagram where

import           Data.Char (toLower)
import           Data.List (sort)

same' :: String -> String -> Bool
same' a b = sort al == sort bl && al /= bl
    where 
        al = map toLower a
        bl = map toLower b


anagramsFor :: String -> [String] -> [String]
anagramsFor s = filter (same' s)
